koschei/models.py

```diff
@@ -611,6 +611,23 @@ class AdminNotice(Base):
     content = Column(String, nullable=False)
 
 
+class ActionLog(Base):
+    id = Column(Integer, primary_key=True)
+    user_id = Column(
+        Integer,
+        ForeignKey('user.id', ondelete='SET NULL'),
+        nullable=True,
+    )
+    user = relationship('User')
+    environment = Column(
+        Enum('admin', 'backend', 'frontend', name='action_log_environment'),
+        nullable=False
+    )
+    timestamp = Column(DateTime, nullable=False,
+                       server_default=func.clock_timestamp())
+    message = Column(String, nullable=False)
+
```

Comment:
I'm wondering if it wouldn't make sense to add reference to BasePackage too. It would be useful to be able to see history for given package.

Reply:
Sometimes it references base_package, sometimes package, other times other tables. What about using a hstore to store arbitrary references?

Reply:
What other tables you have in mind? I think base_package would work fine - are there any cases in which you know package id, but not base_package id?

Reply:
> What other tables you have in mind?

PackageGroup

> are there any cases in which you know package id, but not base_package id?
No, but you're losing a bit of information (in which collection was it). But maybe I should instead include the collection in the message.